### ISSUE TITLE
How to avoid doubled diagnostics ruff vs. ty?

### ISSUE THREAD

**Issue by jamilraichouni** [astral-sh/ty#2195]
### Question

Using Neovim v0.11.5 and `ruff` v0.14.10 I get this:

<img width="1357" height="411" alt="Image" src="https://github.com/user-attachments/assets/2975e858-b1c8-4171-8551-e67e43dd195b" />

My config is at the top right hand side and I marked the diagnostic error message which appears twice.

I looked at https://github.com/astral-sh/ty/blob/8c54a1cca2c41414bd4bc56c3b7b0591007f6d0f/docs/reference/editor-settings.md and also scanned the issues first.

Many thanks,
Jamil

### Version

0.0.6

**Comment by carljm** [comment]
I'm more familiar with the ty side than the ruff side, but on the ty side you can just disable the `invalid-syntax` rule if you are also running ruff, e.g. in your `pyproject.toml`:

```toml
[tool.ty.rules]
invalid-syntax = "ignore"
```

**Comment by AlexWaygood** [comment]
> I'm more familiar with the ty side than the ruff side, but on the ty side you can just disable the `invalid-syntax` rule if you are also running ruff, e.g. in your `pyproject.toml`:
> 
> [tool.ty.rules]
> invalid-syntax = "ignore"

I don't think that's true @carljm. A decision was made that `invalid-syntax` should be different from all other error codes and should not be suppressable via the usual mechanism.

There is however a ruff setting to switch off reporting of syntax errors in the editor specifically: https://docs.astral.sh/ruff/editors/settings/#showsyntaxerrors.

And there is an open issue to add the same setting for ty: https://github.com/astral-sh/ty/issues/2106

**Comment by jamilraichouni** [comment]
Thanks!

Tried that via

```lua
vim.lsp.config("ty", {
    cmd = { "ty", "server" },
    filetypes = { "python" },
    root_markers = { "pyproject.toml", ".git", "ty.toml" },
    settings = {
        -- https://github.com/astral-sh/ty/blob/main/docs/reference/editor-settings.md
        ty = {
            disableLanguageServices = true,
            configuration = {
                rules = {
                    ["invalid-syntax"] = "ignore",
                },
            }
        }
    }
})
```

and the diagnostic still appears.
#2106 sounds like what I need.

**Comment by carljm** [comment]
Seems like you could probably set `showSyntaxErrors` to `false` in your Ruff LSP config today, if you don't want to wait for #2106 in ty?
